### ISSUE TITLE
Nonce wrong for client.create_order

### ISSUE THREAD

**Issue by PencilBow** [sammchardy/python-idex#3]
Hi.

I'm trying to use the api as follows:
```
client = Client("public_key", "private_key")
r = client.create_order(
    token_buy = "EOS",   # The name or address of the token you will receive as a result of the trade
    token_sell = "ETH",   # The name or address of the token you will lose as a result of the trade
    price = eth_price,    # The price in token_sell you want to purchase the new token for
    quantity = amount   # The amount of token_buy you will receive when the order is fully filled
)
```

which receives an error from idex-API:
```
IdexAPIException: IdexAPIException: ER_WARN_DATA_OUT_OF_RANGE: Out of range value for column 'nonce' at row 1
```

Looking in to the python-idex source-code, it seems `_get_nonce` is used to create the nonce for `create_order`
```
    def _get_nonce(self):
        """Get a unique nonce for request
        """
        return int(time.time() * 1000)
```

which seems to contradict
https://github.com/AuroraDAO/idex-api-docs
```
All hashes which are to be signed require a nonce parameter to derive, which must be a numeric value 
that increases by 1 with every message that requires a nonce. 
... 
A nonce must increase by 1 for every API call that requires one, 
i.e. you cannot use the same nonce for an order and then a withdrawal.
```

I saw `get_my_next_nonce` exists, but `create_order` doesnt seems to use that one. 

Am I using the API in a wrong way? 
Any help would be much appreciated :)

**Comment by sammchardy** [comment]
Hi @PencilBow no you aren't doing anything wrong.

The nonce value sent was too high, so I've modified how that is generated in v0.2.1.

**Comment by PencilBow** [comment]
Thanks for the speedy reply & fix, buy order worked. Cancelling an order still seems to generate a wrong nonce.
```
orders = client.get_my_open_orders('ETH_EOS')
...
client.cancel_order("<--orderHash-->")
IdexAPIException: IdexAPIException: Cancel signature invalid. Please try again.
```

the client is providing the following:
`['nonce', 26740, 'uint256']`
however, `client.get_my_next_nonce()` returns `{'nonce': 17430}`

**Comment by PencilBow** [comment]
I managed to fix it myself:
```

    def _request(self, method, path, signed, **kwargs):
        """
        kwargs is `json` or `hash_data`, or both

        example:
            hash_data = [
                ['tokenBuy', buy_currency['address'], 'address'],
                ['amountBuy', amount_buy, 'uint256'],
            ]
            json = {
                'address': self._wallet_address
            }
        """

        if signed:
            # generate signature from hash_data
            # eg {'v': 28 (or 27), 'r': '0x...', 's': '0x...'}
            kwargs['json'].update(self._generate_signature(kwargs['hash_data']))


            # put hash_data also in json param
            for name, value, _param_type in kwargs['hash_data']:
                print(name, value, _param_type)
                kwargs['json'][name] = value
            # kwargs = {
            #     'hash_data': [ ['amountBuy', 150, 'uint256'] ],
            #     'json': {
            #         'v': 27,
            #         'r': '0x1234...'
            #         's': '0x1234...'
            #         'amountBuy': 150,
            #     }
            # }

            # filter out contract address, not required
            if 'contract_address' in kwargs['hash_data']:
                del kwargs['json']['contract_address']


            # remove the passed hash data
            del kwargs['hash_data']
            # kwargs = {
            #     'json': {
            #         'v': 27,
            #         'r': '0x1234...'
            #         's': '0x1234...'
            #         'amountBuy': 150,
            #     }
            # }

        uri = f'{self.API_URL}/{path}'
        response = getattr(self.session, method)(uri, **kwargs)
        return self._handle_response(response)
```

```
    def cancel_order(self, order_hash):
        """Cancel an order

        :param order_hash: The raw hash of the order you are cancelling
        :type order_hash:

        .. code:: python

            status = client.cancel_order('0xcfe4018c59e50e0e1964c979e6213ce5eb8c751cbc98a44251eb48a0985adc52')

        :returns: API Response

        .. code-block:: python

            {
                success: 1
            }

        :raises:  IdexWalletAddressNotFoundException, IdexPrivateKeyNotFoundException, IdexResponseException,  IdexAPIException

        """

        if not self._wallet_address:
            raise IdexWalletAddressNotFoundException()

        if not self._private_key:
            raise IdexPrivateKeyNotFoundException()

        hash_data = [
            ['orderHash', order_hash, Type.address],
            ['nonce', self._get_nonce(), Type.uint256],
        ]

        json_data = {
            'address': self._wallet_address
        }


        return self._post('cancel', True, hash_data=hash_data, json=json_data)
```

**Comment by sammchardy** [comment]
Thanks @PencilBow, overlooked that in the docs.

I've released v0.2.2 with that change.
